### DISCUSSION TITLE
How to use the second parameter of `Painter::arrow`?

### DISCUSSION THREAD

**Discussion by amandasystems** [emilk/egui#3520] [Q&A] [answered]
The documentation is not clear on what the meaning of the second parameter is. While the length is fairly sraightforward (it's `vec.y` I guess?), it's less clear about the unit of the direction, and I can't figure out how to make a `Vec2` of the length I want plus an angle.

I guess this is straightforward if you're into linear algebra, but I'm not.

**Comment by JumpyLionnn** [comment, marked as answer]
The second parameter means both in what direction and what length the arrow will go. 
In linear algebra a vector has a direction and a length. it can be represented by an angle and a length scalar or by an x and y.
The head of the arrow will just be at the position of the sum of the 2 vectors
Here is an example, if the origin is (2, 1) and the second parameter is (4, 6) then
there will be an arrow between (2, 1) and (2+ 4, 6 + 1) = (6, 7).
If you want to get a vector based on a direction and a length you can do the following:
```rust
let length = 10.0;
// angle is radians
let vec = egui::Vec2::angled(1.0) * length;
```
If you need any help or reexplanation, let me know. I hope it helped you :)

**Comment by amandasystems** [comment]
Thanks! This almost did what I wanted, but it also woke up my slumbering linear algebra and I now remember how to construct vectors between points in space, so now I can actually use this function.
